nx/guides/modoc_agg_nx_01.livemd

```diff
@@ -0,0 +1,826 @@
+
+$$
+\begin{bmatrix} x=0
+  \begin{bmatrix} y=0
+
+\begin{bmatrix}
+  a_{0,0,0,0} & a_{0,0,0,1} & a_{0,0,0,2} \\
+  a_{0,0,1,0} & a_{0,0,1,1} & a_{0,0,1,2} \\
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix} 
+  a_{0,1,0,0} & a_{0,1,0,1} & a_{0,1,0,2} \\
+  a_{0,1,1,0} & a_{0,1,1,1} & a_{0,1,1,2} \\
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  a_{0,2,0,0} & a_{0,2,0,1} & a_{0,2,0,2} \\
+  a_{0,2,1,0} & a_{0,2,1,1} & a_{0,2,1,2} \\
+\end{bmatrix}
+
+\end{bmatrix}
+  \\
+
+x=1
+
+\begin{bmatrix} y=0
+
+\begin{bmatrix}
+  a_{1,0,0,0} & a_{1,0,0,1} & a_{1,0,0,2}  \\
+  a_{1,0,1,0} & a_{1,0,1,1} & a_{1,0,1,2}  \\
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix}
+  a_{1,1,0,0} & a_{1,1,0,1} & a_{1,1,0,2}  \\
+  a_{1,1,1,0} & a_{1,1,1,1} & a_{1,1,1,2}  \\
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  a_{1,2,0,0} & a_{1,2,0,1} & a_{1,2,0,2}  \\
+  a_{1,2,1,0} & a_{1,2,1,1} & a_{1,2,1,2}  \\    
+\end{bmatrix}
+
+\end{bmatrix}
+\end{bmatrix}
+
+
+
+\begin{bmatrix} x=0
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & 1, 2, 3, & \\
+  & 1, -2, 3, &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix} 
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 5, -1, 3 &\\
+  & -5, 1, 3 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+  \\
+  x=1
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & 4, 6, 1 & \\
+  & 4, -6, 1 &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix}
+  & 1, 2, 3 & \\
+  & 1, -2, 3 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+Firstly, let's check some characteristics of this tensor: they use full tensor aggregation functions.
+
+```elixir
+%{
+  dimensions: %{
+    x: Nx.axis_size(t, :x),
+    y: Nx.axis_size(t, :y),
+    z: Nx.axis_size(t, :z),
+    t: Nx.axis_size(t, :t)
+  },
+  n: Nx.size(t),
+  shape: Nx.shape(t),
+  rank: Nx.rank(t),
+  analysis: %{
+    most_frequent_nb: Nx.mode(t) |> Nx.to_number(),
+    smallest_element: %{
+      value: Nx.reduce_min(t) |> Nx.to_number(),
+      position: Nx.argmin(t) |> Nx.to_number()
+    },
+    greatest_element: %{
+      value: Nx.reduce_max(t) |> Nx.to_number(),
+      position: Nx.argmax(t) |> Nx.to_number()
+    },
+    no_zero_nb: Nx.all(t) |> Nx.to_number()
+  }
+}
+```
+
+#### Single row aggregation, along an axis
+
+> ❗ We are going to use the key `:axis` below. The following functions `argmin`, `argmax`, `median` and `mode` use `:axis: <symbol> or number` in singular mode. All of the other aggregating functions use `axes: [<symbol> or number]`; these have multi-axis aggregation implemented.
+
+When you aggregate along an axis, you are going to reshape the tensor and to aggregate with a function, in other words perform a reduction. It is important to consider the _ordering_ of the axes.
```

Comment:
```suggestion
When you aggregate along an axis, it's as if you're selecting all elements of a tensor along that axis for each fixed set of indices in other axes, and then performing the aggregation function over the selected subset.
```

"_ordering_" here is confusing. Do you mean the order in which the axes are originally distributed?